### ISSUE TITLE
Type tokens not working in mqtt block

### ISSUE THREAD

**Issue by cpressman** [taverntesting/tavern#293]
I used the example from the documentation (Sidenote: it uses mqtt as a label instead of paho-mqtt so it should be updated). I made the mqtt-publish and mqtt response both look for ping and it worked. If I changed the mqtt-response to !anything I get the error below. I tried the same thing with json and had the same errors.

### Code:
```yaml
# test_mqtt.tavern.yaml
---

test_name: Test mqtt message response
paho-mqtt:
  client:
    transport: tcp
    client_id: tavern-tester
  connect:
    host: 10.1.0.67
    port: 1883
    timeout: 3

stages:
  - name: step 1 - ping/pong
    mqtt_publish:
      topic: test
      payload: ping
    mqtt_response:
      topic: test
      payload: !anything
      timeout: 5
```

### Error Output (edit: removed < from before "Tavern" on top and bottom line. Was causing the line not to show up).
Got unexpected payload on topic 'test': 'ping' (expected 'Tavern YAML sentinel for anything>')
Message not received after 4 seconds
Expected 'Tavern YAML sentinel for anything>' on topic 'test' but no such message received

**Comment by cpressman** [comment]
Just to update:
###Works
json:
  ping: ping
###Fails
json:
  ping: !anything

Got unexpected payload on topic 'test': '{'ping': 'ping'}' (expected '{'ping': <tavern.util.loader.AnythingSentinel object at 0x000001E6E5BD50F0>}')
Message not received after 4 seconds
Expected '{'ping': <tavern.util.loader.AnythingSentinel object at 0x000001E6E5BD50F0>}' on topic 'test' but no such message received

**Comment by michaelboulton** [comment]
#295 should fix the first issue but I can't reproduce the second example (see https://github.com/taverntesting/tavern/pull/295/files#diff-683c8743ef7e2fa34becb9dff4db93b7R288)

**Comment by KenStorey** [comment]
The second case is similar to #226 which was fixed under #234.

**Comment by cpressman** [comment]
I checked out #234 and this is fixed there. Thanks! Closing.
